server/src/internalClusterTest/java/org/opensearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java

```diff
@@ -188,6 +186,89 @@ public void testHighWatermarkNotExceeded() throws Exception {
         assertBusyWithDiskUsageRefresh(dataNode0Id, indexName, hasSize(1));
     }
 
+    public void testIndexCreateBlockWhenAllNodesExceededHighWatermark() throws Exception {
+        internalCluster().startClusterManagerOnlyNode();
+        final List<String> dataNodeNames = internalCluster().startDataOnlyNodes(2);
+        ensureStableCluster(3);
+
+        final InternalClusterInfoService clusterInfoService = (InternalClusterInfoService) internalCluster()
+            .getCurrentClusterManagerNodeInstance(ClusterInfoService.class);
+        internalCluster().getCurrentClusterManagerNodeInstance(ClusterService.class).addListener(event -> clusterInfoService.refresh());
+
+        // Reduce disk space of all node until all of them is breaching high disk watermark.
+        for (final String dataNodeName : dataNodeNames) {
+            populateNode(dataNodeName);
+        }
+
+        // Validate if cluster block is applied on the node
+        assertBusy(() -> {
+            refreshDiskUsage();
+            ClusterState state = client().admin().cluster().prepareState().setLocal(true).get().getState();
+            assertTrue(state.blocks().hasGlobalBlockWithId(DiskThresholdMonitor.INDEX_CREATE_BLOCK_ID));
```

Comment:
This is already tested in CreateIndexBlockIt: https://github.com/RS146BIJAY/OpenSearch/blob/bd99a2e75d8ad282f8c0205945cc01456e39b6db/server/src/internalClusterTest/java/org/opensearch/blocks/CreateIndexBlockIT.java#L22. If index create block is present, it will not allow index creation. I think it will be redundant if we check this again here.